### PR TITLE
Polling updates

### PR DESCRIPTION
**Pull request by niquerio**
Does this stuff: 

- [x] Button to publish poll results to the order
- [x] Scheduler to process poll if there's a poll that needs to processed and publish it for the royals
- [x] Generate submission analytics for order secretary so she can know who has or hasn't completed their poll yet
- [x] Improve the comment view for royalty
- [x] Generate Offline poll
- [x] Process for Order Secretary to submit an offline poll

I'm mostly concerned with the code duplication for Watch List and Vote List candidates.

### CODE REVIEW

File: app/models/advising.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
   belongs_to :candidate
-  belongs_to :poll, optional: true
+  belongs_to :poll
   
   enum judgement: [:elevate, :wait, :no_strong_opinion, :drop], _prefix: true
+  def date
+    poll.date 
+  end
   
   def self.vote_list_judgements
```
[the remark this answers is not in the document]
Comment (aelkiss) [reply app/models/advising.rb]: You could consider extracting `vote_list_judg~e~ments` and friends to separate presenter classes for vote list advisings & watch list advisings

File: app/services/offline_poll_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class OfflinePollPresenter
+  def initialize(peerage)
+    @peerage = peerage
+    @candidates = Candidate.where_order(@peerage)
+	end
+  def vote_list
+    @candidates.where(vote:true).order(:sca_name).map { |c| OfflinePollCandidatePresenter.new(c) }
```
Comment (aelkiss) [inline app/services/offline_poll_presenter.rb]: extract method?

File: app/views/chambers/peerage/admin/offline_poll_submitters/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<div class="row justify-content-center">
+  <div class="col-md-8">
+    <h1>Submit Offline <%=peerage.capitalize %> Poll</h1>
+    <%= bootstrap_form_for @ballot, url: chambers_admin_offline_poll_submitter_path do |f| %>
+      <%= f.select :peer_id, @ballot.peers.map{|p| [p.sca_name, p.id] }, {label: 'Peers', include_blank: true}, {class: 'selectpicker', data: {'live-search' => true}, required: true} %> 
+      <h3>Vote List</h3>
+      <%= f.fields_for :vote_list_advisings do |adv_field| %>
```
Comment (aelkiss) [inline app/views/chambers/peerage/admin/offline_poll_submitters/new.html.erb]: Consider `f.fields_for :vote_list_advisings.select { |a| a.object.candidate.vote_list }` or something like that here, and the opposite below. The fields should ultimately have the same names (you can verify that in the generated HTML)

File: app/services/offline_ballot.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    Peer.where_order(@peerage).where(active: true).select{ |p| !peer_has_started_poll?(p) }
+  end
+	def watch_list_advisings_attributes=(attributes)
+		attributes.each do |i, adv_params|
+			adv_params["peer_id"] = @peer_id
+			my_adv = @watch_list_advisings.find{|adv| adv.candidate_id == adv_params["candidate_id"].to_i}
+			my_adv.assign_attributes(adv_params)
+		end	
+	end
+	def vote_list_advisings_attributes=(attributes)
```
Comment (aelkiss) [inline app/services/offline_ballot.rb]: Collapse this to just `advisings_attributes=` if going to a filtered `fields_for :advisings` below works

File: app/services/offline_ballot.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+		value.each do |adv|
+			record.errors.add attr, 'every candidate needs judgement' if adv.judgement.blank? 
+		end
+	end
+	
+	attr_accessor :vote_list_advisings, :peer_id, :watch_list_advisings
+  def initialize(peerage)
+    @peerage = peerage 
+		@poll = Poll.active_for(@peerage)
+		@vote_list_advisings = Candidate.where_order(@peerage).where(vote:true).order(:sca_name).map{ |c| Advising.find_or_initialize_by(candidate: c, poll: @poll, peer_id: @peer_id )}
```
[the remark this answers is not in the document]
Comment (niquerio) [reply app/services/offline_ballot.rb]: Added a scope on Candidate. Am not going to do it on advisings unless it comes up again.

File: app/services/offline_poll_presenter.rb (hunk 2 above)
Comment (niquerio) [reply app/services/offline_poll_presenter.rb]: I'm not going to change this now, but I will if I have a third kind of list or if I have to do this again in the future

**Pull request merged**